File: tmp-mei/fgen4255027576711118260/base/caps/Read_0.java
package base.caps;
public interface Read_0 extends base.caps.FileHandleMode_0{
Read_0 $self = new Read_0Impl();
rt.Str str$read();
static rt.Str str$read$fun(base.caps.Read_0 $this) {
  return ((rt.Str)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (rt.Str) null;
}})
);
}
}

File: tmp-mei/fgenjs5030319945734616766/base/caps/Read_0.js
import { base$$caps$$FileHandleMode_0 } from "../../base/caps/FileHandleMode_0.js";
import * as rt from "../rt/main.js";

export class base$$caps$$Read_0 extends base$$caps$$FileHandleMode_0 {
  static $self = new base$$caps$$Read_0();

  str$read() {
  return base$$caps$$Read_0.str$read$fun(this);
}

  static str$read$fun($this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}
